packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -504,10 +504,13 @@ export class DidExchangeProtocol {
     message: DidExchangeRequestMessage | DidExchangeResponseMessage,
     invitationKeysBase58: string[] = []
   ) {
-    // The only supported case where we expect to receive a did-document attachment is did:peer algo 1
-    return isDid(message.did, 'peer') && getNumAlgoFromPeerDid(message.did) === PeerDidNumAlgo.GenesisDoc
-      ? this.extractAttachedDidDocument(agentContext, message, invitationKeysBase58)
-      : this.extractResolvableDidDocument(agentContext, message, invitationKeysBase58)
+    // Not all agents use didRotate yet, some may still send a didDoc attach with various did types
+    // we should check if the didDoc attach is there and if not require that the didRotate be present
+    if (message.didDoc) {
+      return this.extractAttachedDidDocument(agentContext, message, invitationKeysBase58)
+    } else {
+      return this.extractResolvableDidDocument(agentContext, message, invitationKeysBase58)
+    }
```

Comment:
Sorry to be so nitpicky and pedandit, but I want to be careful with this, as it can have a lot of impact on security. 

If we allow any did document to be passed, it means that e.g. for a did:web document, I could provide it as attachment heren, and how this is now coded we will store the document as is. That means I could impersonate a did:web document, as we now don't need to resolve it. This means we bypass the ledger security layer most dids provide.

So I think we should implement the logic as follows:
- check if the did is resolvable using a did resolver (all dids except for did:peer:1)
  - if it is resolvable, and the did is the same as in the invitation we will resolve the DID, without requireing `did_rotate~attach` or `did_doc~attach`.
  - if it is resolvable, and the did is different from the did in the invitation, verify the `did_rotate~attach` has been signed with the invitation did (as we do now) and resolve the did
  - if it is not resolvable (so did:peer:1), we will extract the did from the `did_doc~attach` and verify the signature

I would be curious: in ACA-Py do you use `did_doc~attach` also with did:peer:4?